validator/src/main.rs

```diff
@@ -1219,17 +1267,25 @@ pub fn main() {
     }
 
     if let Some(ref cluster_entrypoint) = cluster_entrypoint {
-        let mut udp_sockets = vec![
-            &node.sockets.gossip,
-            &node.sockets.repair,
-            &node.sockets.serve_repair,
-        ];
-        udp_sockets.extend(node.sockets.tpu.iter());
-        udp_sockets.extend(node.sockets.tpu_forwards.iter());
-        udp_sockets.extend(node.sockets.tvu.iter());
-        udp_sockets.extend(node.sockets.tvu_forwards.iter());
-        udp_sockets.extend(node.sockets.broadcast.iter());
-        udp_sockets.extend(node.sockets.retransmit_sockets.iter());
+        let mut udp_sockets = vec![&node.sockets.gossip, &node.sockets.repair];
+
+        if ContactInfo::is_valid_address(&node.info.serve_repair) {
+            udp_sockets.push(&node.sockets.serve_repair);
+        }
+        if ContactInfo::is_valid_address(&node.info.tpu) {
+            udp_sockets.extend(node.sockets.tpu.iter());
+        }
+        if ContactInfo::is_valid_address(&node.info.tpu_forwards) {
+            udp_sockets.extend(node.sockets.tpu_forwards.iter());
+        }
+        if ContactInfo::is_valid_address(&node.info.tvu) {
+            udp_sockets.extend(node.sockets.tvu.iter());
+            udp_sockets.extend(node.sockets.broadcast.iter());
+            udp_sockets.extend(node.sockets.retransmit_sockets.iter());
+        }
+        if ContactInfo::is_valid_address(&node.info.tvu_forwards) {
+            udp_sockets.extend(node.sockets.tvu_forwards.iter());
+        }
 
         let mut tcp_listeners = vec![];
         if !private_rpc {
```

Comment:
:eyes: (So, #12239 isn't needed for v1.2 because v1.2 wasn't affected due to #12191 backporting whim. :)